src/core/backbone_router/bbr_manager.cpp

```diff
@@ -90,18 +111,63 @@ void Manager::HandleMulticastListenerRegistration(const Coap::Message &aMessage,
     // TODO: (MLR) send configured MLR response for Reference Device
     // TODO: (MLR) handle Commissioner Session TLV
     // TODO: (MLR) handle Timeout TLV
-    // TODO: (MLR) handle addresses, updated listener groups, send BMLR.req
+
+    VerifyOrExit(ThreadTlv::FindTlvValueOffset(aMessage, IPv6AddressesTlv::kIPv6Addresses, addressesOffset,
+                                               addressesLength) == OT_ERROR_NONE,
+                 error = OT_ERROR_PARSE);
+    VerifyOrExit(addressesLength % sizeof(Ip6::Address) == 0, error = OT_ERROR_PARSE);
+    VerifyOrExit(addressesLength / sizeof(Ip6::Address) <= kIPv6AddressesNumMax, error = OT_ERROR_PARSE);
+
+    SuccessOrExit(error = Get<BackboneRouter::Leader>().GetConfig(config));
+    expireTime = TimerMilli::GetNow() + TimeMilli::SecToMsec(config.mMlrTimeout);
+
+    for (uint16_t offset = 0; offset < addressesLength; offset += sizeof(Ip6::Address))
+    {
+        bool failed = true;
+
+        IgnoreReturnValue(aMessage.Read(addressesOffset + offset, sizeof(Ip6::Address), &address) ==
+                          sizeof(Ip6::Address));
```

Comment:
nit: the `==` should be unnecessary?

```suggestion
        IgnoreReturnValue(aMessage.Read(addressesOffset + offset, sizeof(Ip6::Address), &address));
```